glean-core/android/src/main/java/mozilla/telemetry/glean/Glean.kt

```diff
@@ -217,9 +213,9 @@ open class GleanInternalAPI internal constructor () {
                 ProcessLifecycleOwner.get().lifecycle.addObserver(gleanLifecycleObserver)
             }
 
-            if (!uploadEnabled) {
-                DeletionPingUploadWorker.enqueueWorker(applicationContext)
-            }
+            // Enqueue the PingUploadWorker, this does not depend on wether upload is enabled,
+            // because there might be deletion-request ping in that case.
+            PingUploadWorker.enqueueWorker(applicationContext)
```

Comment:
How can we make sure to attempt to upload only *deletion-request* pings?

Reply:
The rust part is the one making sure of that now.

Reply:
Do we have a test for that?

Reply:
The is a test for that in sample-app: https://github.com/mozilla/glean/blob/master/samples/android/app/src/androidTest/java/org/mozilla/samples/gleancore/pings/DeletionRequestPingTest.kt

Reply:
The test does not check that the only uploaded ping is the deletion request, but it indeed checks that the specific ping is received :) I think we should either add a new test (not necessarily as part of this bug) or extend that one to make sure that we never upload pings other than the deletion one.

Reply:
https://bugzilla.mozilla.org/show_bug.cgi?id=1620673